[vtk] Fix warnings in VTK::CommonCore

vtkVariantInlineOperators: -Wlogical-op

This patch has been upstreamed and can likely be dropped in a future
update to Drake's VTK.

See https://gitlab.kitware.com/vtk/vtk/-/merge_requests/10773.

--- Common/Core/vtkVariantInlineOperators.h
+++ Common/Core/vtkVariantInlineOperators.h
@@ -14,7 +14,9 @@
 VTK_ABI_NAMESPACE_BEGIN
 inline bool IsSigned64Bit(int VariantType)
 {
-  return ((VariantType == VTK_LONG_LONG) || (VariantType == VTK_TYPE_INT64));
+  if (VariantType == VTK_TYPE_INT64) return true;
+  if (VariantType == VTK_LONG_LONG) return true;
+  return false;
 }
 
 inline bool IsSigned(int VariantType)
